### DISCUSSION TITLE
DNS Time is Always shows zero in Real User Monitoring

### DISCUSSION THREAD

**Discussion by AjinThankachan** [apache/skywalking#12216] [Q&A]
Hi,
In the Skywalking Demo UI the DNS time is always shows zero. I have tried to to work on my own environment as well the result is same for RUM monitoring using the client JS. Can you tell me is it a bug or did we miss somehing? Thanks for the help in advance.
![Uploading image.png…]()

**Comment by wu-sheng** [comment]
Can't see your screenshot.

**Reply by AjinThankachan** [reply]
Sorry i will try to upload again.In the demo instance this is the link :http://demo.skywalking.apache.org/dashboard/BROWSER/Endpoint/YWdlbnQ6OnVp.1/YWdlbnQ6OnVp.1_L2hvbWVwYWdl/Browser-Page 
![Screenshot 2024-05-13 094936](https://github.com/apache/skywalking/assets/25667606/55346931-e35b-404d-b7b7-da9c0b087ba3)


In Page Performace dns_avg always shows zero.

**Reply by wu-sheng** [reply]
@Fine0830 Where do we get dns metrics?

@AjinThankachan I think you could follow the above answer and check your env. AFAIK, DNS is not doing for every request.

**Reply by AjinThankachan** [reply]
Hi ,

dnsTime:parseInt(String(t.domainLookupEnd-t.domainLookupStart),10)
Is the code that is using the dnTime fetching in Cleint Js i guess.  And in the types:
export type IPerfDetail = {
  redirectTime: number | undefined; // Time of redirection
  dnsTime: number | undefined; // DNS query time

**Reply by AjinThankachan** [reply]
@Fine0830 can you help ?
